app/views/grids/_notes.html.erb

```diff
@@ -74,9 +74,21 @@
 
 <script>
 (function(){
-  $(".<%= className %>-<%= randomSeed %> .show-all a").click(function() {
-    $(".<%= className %>-<%= randomSeed %> tr.hide").toggleClass("hide");
-    $("#notes-table-footer")[0].innerHTML="";
+  $(".<%= className %>-<%= randomSeed %> .show-all a").click(function(event) {
+    var cName = ".<%= className %>-<%= randomSeed %>";
+    var sName = cName + " tr.extra";
+    
+    $(sName).toggleClass("hide");
+    if($(sName)[0].classList.contains("hide"))
+        var hiding = true;    
+    var myElement = event.target;    
+    if(event.target.classList.contains("caret")) 
+        myElement = event.target.parentElement;
+    if(!hiding) 
+        myElement.innerHTML = 'Show <%= nodes.length - 10 %> less <b class="caret" style = "transform: rotate(180deg);"></b>';
```

Comment:
I think it's fine just to use "Show more" and "Show less".

Reply:
Also, I like that you thought about the caret icon too. One more thing could be done changing the mouse cursor to hand on hovering 'show more' link.